bulk-ingester/src/main/java/org/alfresco/hxi_connector/bulk_ingester/processor/mapper/AlfrescoPropertyMapper.java

```diff
@@ -0,0 +1,201 @@
+import org.alfresco.elasticsearch.db.connector.model.ContentMetadata;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValue;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValueType;
+
+@Slf4j
+@RequiredArgsConstructor
+public class AlfrescoPropertyMapper
+{
+    private final AlfrescoNode alfrescoNode;
+    private final String propertyName;
+
+    public Optional<Map.Entry<String, Serializable>> performMapping()
+    {
+        /* Properties might be duplicated - for example they can have different locale (we can have two descriptions with "en_US_" and "en_UK_" locale) and in this case we want to process them together */
+        List<Serializable> propertyValues = alfrescoNode.getNodeProperties().stream()
+                .filter(nodeProperty -> nodeProperty.getPropertyKey().getLocalName().equals(propertyName))
+                .map(nodeProperty -> mapPropertyValue(nodeProperty.getPropertyValue()))
+                .flatMap(Optional::stream)
+                .toList();
+
+        if (propertyValues.isEmpty())
+        {
+            return empty();
+        }
+        else if (propertyValues.size() == 1)
+        {
+            return of(Map.entry(propertyName, propertyValues.get(0)));
+        }
+        else
+        {
+            return of(Map.entry(propertyName, (Serializable) propertyValues));
+        }
+    }
+
+    private Optional<Serializable> mapPropertyValue(PropertyValue propertyValue)
+    {
+        try
+        {
+            return getPersistedValue(propertyValue).flatMap(persistedValue -> switch (PropertyValueType.getPropertyValueType(propertyValue.getActualType()))
+            {
+            case CONTENT_DATA_ID -> getContentValue(persistedValue);
+            case DATE -> getDateValue(persistedValue);
+            case NODEREF -> getNodeRefValue(persistedValue);
+            default -> of(persistedValue);
+            });
+        }
+        catch (Exception e)
+        {
+            log.warn(format("Error occurred while trying to map property %s value %s of node %s", propertyName, propertyValue, alfrescoNode.getId()), e);
+
+            return empty();
+        }
+    }
+
+    private Optional<Serializable> getNodeRefValue(Serializable persistedValue)
+    {
+        try
+        {
+            String stringValue = (String) persistedValue;
+            String[] protocolSplit = stringValue.split("://");
+
+            String protocol = protocolSplit[0];
+
+            String[] storeSplit = protocolSplit[1].split("/");
+            String storeIdentifier = storeSplit[0];
+            String id = storeSplit[1];
+
+            return of((Serializable) Map.of(
+                    "id", id,
+                    "storeRef", Map.of("identifier", storeIdentifier, "protocol", protocol)));
+        }
+        catch (Exception e)
+        {
+            log.error(format("Cannot deserialize noderef property %s with value %s for node %s", propertyName, persistedValue, alfrescoNode.getId()), e);
```

Comment:
Slf4j already has token handling:
```suggestion
            log.error("Cannot deserialize noderef property {} with value {} for node {}", propertyName, persistedValue, alfrescoNode.getId(), e);
```

Reply:
fixed